audit/audit-org-usage.py

```diff
@@ -0,0 +1,335 @@
+    def __init__(self, arn, tags):
+        super().__init__(arn, tags)
+
+
+class Es(AWSNotS3):
+    def __init__(self, arn, tags):
+        super().__init__(arn, tags)
+
+    def get_es_instance(self, client):
+        es_domain_name = self.arn.split("/")[1]
+        response = client.describe_elasticsearch_domain(DomainName=es_domain_name)
+        domain_status = response["DomainStatus"]
+        ebs_options = domain_status.get("EBSOptions", {})
+        self.volume_size = ebs_options.get("VolumeSize", 0)
+
+
+class S3(AWSResource):
+    def __init__(self, arn, tags):
+        super().__init__(arn, tags)
+        self.bucket_name = self.instance_id
+
+    def get_s3_usage(self, client):
+        now = datetime.datetime.now()
+        response = client.get_metric_statistics(
+            Namespace="AWS/S3",
+            MetricName="BucketSizeBytes",
+            Dimensions=[
+                {"Name": "BucketName", "Value": self.bucket_name},
+                {
+                    "Name": "StorageType",
+                    "Value": "StandardStorage",
+                },
+            ],
+            Statistics=["Average"],
+            Period=86400,
+            StartTime=now - datetime.timedelta(days=1),
+            EndTime=now,
+            Unit="Bytes",
+        )
+        self.s3_usage = 0
+        datapoints = response["Datapoints"]
+        if len(datapoints) > 0:
+            self.s3_usage = datapoints[0]["Average"]
+
+
+class Organization:
+    def __init__(self, name):
+        self.name = name
+        self.data = self.get_data()
+        self.guid = self.data["guid"]
+        self.quota_guid = self.data["relationships"]["quota"]["data"]["guid"]
+        self.rds_instances = []
+        self.redis_instances = []
+        self.es_instances = []
+        self.s3_buckets = []
+
+    def get_data(self):
+        cf_json = subprocess.check_output(
+            "cf curl /v3/organizations/?names=" + self.name,
+            universal_newlines=True,
+            shell=True,
+        )
+        return json.loads(cf_json)["resources"][0]
+
+    def get_quota_memory(self):
+        cf_json = subprocess.check_output(
+            "cf curl /v3/organization_quotas/" + self.quota_guid,
+            universal_newlines=True,
+            shell=True,
+        )
+        return json.loads(cf_json)["apps"]["total_memory_in_mb"]
+
+    def get_memory_usage(self):
+        cf_json = subprocess.check_output(
+            "cf curl /v3/organizations/" + self.guid + "/usage_summary",
+            universal_newlines=True,
+            shell=True,
+        )
+        return json.loads(cf_json)["usage_summary"]["memory_in_mb"]
+
+    def get_aws_instances(self, client, resource_type):
+        resource_type_map = {
+            "rds": "rds:db",
+            "redis": "elasticache:replicationgroup",
+            "es": "es:domain",
+        }
+        resource_type_filter = [resource_type_map[resource_type]]
+        tag_key = "Organization GUID"
```

Comment:
It's not _wrong_ to rely on the Organization GUID tag, and while the broker **should be** tagging resources correctly going forward, I never went back to verify that old instances had all been tagged correctly. I'm planning to write reconciliation scripts to handle this use case, but it's on the backlog

Reply:
yeah, I think I can iterate over the all the managed services for an org with something like:

`cf curl '/v3/service_instances?organization_guids=c0b94e73-f74d-4937-8e51-68b290bcc4d2&type=managed&fields%5Bservice_plan.service_offering.service_broker%5D=guid%2Cname&fields%5Bservice_plan.service_offering%5D=guid%2Cname%2Crelationships.service_broker&fields%5Bservice_plan%5D=guid%2Cname%2Crelationships.service_offering'`

and then get the AWS details by the Instance GUID tag (hopefully that's correct!)

Reply:
@jameshochadel here's the discussion of AWS tagging vs getting services from the API.